recipes/xgboost/build.sh

```diff
@@ -0,0 +1,6 @@
+if [ -n "${OSX_ARCH+x}" ]; then
+  patch xgboost/Makefile $RECIPE_DIR/Makefile.patch
+  patch xgboost/build-python.sh $RECIPE_DIR/build-python.patch
```

Comment:
`conda-build` has a mechanism for doing this already in the [`patches`]( https://conda.io/docs/building/meta-yaml.html#patches ) section of `source`. Please update to use that.